core/core.go

```diff
@@ -565,3 +565,22 @@ func getData[T any](api *DefaultAPI, token string, url string) (T, error) {
 
 	return data, nil
 }
+
+const maxBytesLimit int64 = 1024 * 1024 * 10 // 10MB
+
+// re-implementation of io.ReadAll with max limit
+func MaxBytesReadAll(r io.Reader) ([]byte, error) {
+	limitedReader := &io.LimitedReader{
+		R: r,
+		N: maxBytesLimit,
+	}
+	data, err := io.ReadAll(limitedReader)
+	if err != nil {
+		return nil, err
+	}
+	if limitedReader.N == 0 {
```

Comment:
Isn't this off-by-one here? So you can't read 10MB of data now, because you'll have `N` set to 0 and it will fail, right? :thinking:

Reply:
that's right, there was a bug related to off-by-one, but not there.

I've created tests that prove that this is working now with input length of -1 +0 or +1 in difference to the limit